core/src/main/java/com/graphhopper/reader/dem/HeightTile.java

```diff
@@ -36,21 +36,27 @@
     private final int minLat;
     private final int minLon;
     private final int width;
-    private final int degree;
+    private final int height;
+    private final int horizontalDegree;
+    private final int verticalDegree;
     private final double lowerBound;
-    private final double higherBound;
+    private final double lonHigherBound;
+    private final double latHigherBound;
     private DataAccess heights;
     private boolean calcMean;
 
-    public HeightTile(int minLat, int minLon, int width, double precision, int degree) {
+    public HeightTile(int minLat, int minLon, int width, int height, double precision, int horizontalDegree, int verticalDegree) {
```

Comment:
The GMTED tifs are not quadratic, the same is true for other elevation sources as well, so it is quite important that we can handle non quadratic files.